### DISCUSSION TITLE
How to monitor healing activities

### DISCUSSION THREAD

**Discussion by SoftBoil** [minio/minio#11664] [Q&A] [answered]
Can someone explain how I can monitor the background healing activities?  In particular I'd like to know when bitrot is detected.

I have Prometheus and Grafana setup but the self-healing stats never seem to go up (eg. self_heal_objects_scanned remain at "1" over the last 24 hrs). There is alot of idle time on the server so I expect the background healing to be frequently scanning objects.  Are the Prometheus stats only related to the deprecated manual heal command?  If so, how can I monitor the new background healing?

**Comment by harshavardhana** [comment, marked as answer]
Bit-rot is not run on a regular basis and by default it's disabled you need to turn on heal config to enable bit-rot, notice enabling bit-rot on a constant basis might just make CPU usage spike.

Also bit-rot is verified everytime you read an object and we find bit-rot healing is triggered automatically.

**Reply by SoftBoil** [reply]
Thanks for the reply.  
I do have the setting bitrotscan=on.  Are there stats or notification for when bit-rot is detected?

**Reply by harshavardhana** [reply]
> I do have the setting bitrotscan=on. Are there stats or notification for when bit-rot is detected?

No there are no such stats right now, heal visibility is being improved here https://github.com/minio/minio/pull/11312
